server/src/main/java/io/crate/execution/engine/aggregation/impl/average/numeric/NumericAverageAggregation.java

```diff
@@ -197,150 +184,11 @@ public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver refer
-                                                                          MemoryManager memoryManager,
-                                                                          Version minNodeVersion) {
-            ramAccounting.addBytes(INIT_SIZE);
-            return new NumericAverageState<>(new OverflowAwareMutableLong(0L), 0L);
-        }
-
-        @Override
-        public void loadDocValues(LeafReaderContext reader) throws IOException {
-            values = DocValues.getSortedNumeric(reader.reader(), columnName);
-        }
-
-        @Override
-        public void apply(RamAccounting ramAccounting,
-                          int doc,
-                          NumericAverageState<OverflowAwareMutableLong> state) throws IOException {
-            if (values.advanceExact(doc) && values.docValueCount() == 1) {
-                if (state != null) {
-                    state.sum.add(values.nextValue());
-                    state.count++;
-                }
-            }
-        }
-
-        @Override
-        public Object partialResult(RamAccounting ramAccounting, NumericAverageState<OverflowAwareMutableLong> state) {
-            return state;
-        }
-    }
-
-    static class AvgDouble implements DocValueAggregator<NumericAverageState<BigDecimalValueWrapper>> {
-
-        private final DataType<BigDecimal> returnType;
-        private final String columnName;
-        private SortedNumericDocValues values;
-
-        AvgDouble(DataType<BigDecimal> returnType, String columnName) {
-            this.returnType = returnType;
-            this.columnName = columnName;
-        }
-
-        @Override
-        public NumericAverageState<BigDecimalValueWrapper> initialState(RamAccounting ramAccounting,
-                                                                        MemoryManager memoryManager,
-                                                                        Version minNodeVersion) {
-            ramAccounting.addBytes(INIT_SIZE);
-            return new NumericAverageState<>(new BigDecimalValueWrapper(BigDecimal.ZERO), 0L);
-        }
-
-        @Override
-        public void loadDocValues(LeafReaderContext reader) throws IOException {
-            values = DocValues.getSortedNumeric(reader.reader(), columnName);
-        }
-
-        @Override
-        public void apply(RamAccounting ramAccounting,
-                          int doc,
-                          NumericAverageState<BigDecimalValueWrapper> state) throws IOException {
-            if (values.advanceExact(doc) && values.docValueCount() == 1) {
-                if (state != null) {
-                    BigDecimal value = returnType.implicitCast(NumericUtils.sortableLongToDouble(values.nextValue()));
-                    BigDecimal newValue = state.sum.value().add(value);
-                    ramAccounting.addBytes(NumericType.sizeDiff(newValue, state.sum.value()));
```

Comment:
RamAccounting was done here which is now missing, right? 
Could we adapt the `SortedNumericDocValueAggregator` and the `BinaryDocValueAggregator` to pass the RamAccounting to it's consumer to that we can account for it in the implementations?

Reply:
Added in a separate PR, and I will rebase on top of it this one once merged. thank you!
https://github.com/crate/crate/pull/17966